backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/gson/impl/AbstractConfiguredExtraPropertiesJsonAdapterFactory.java

```diff
@@ -115,7 +115,7 @@ private static <T> Constructor<T> getConstructor(TypeToken<T> type) {
 		try {
 			return (Constructor<T>) type.getRawType().getConstructor();
 		}
-		catch (NoSuchMethodException | SecurityException e) {
```

Comment:
References to `SecurityException` do not need to be removed, because:

1. It's a runtime exception, so our code will compile even if the code inside the try block does not declare throwing that exception.
2. `SecurityException` is not planned for removal at the moment:
   > We will not deprecate some classes in the java.security package that are related to the Security Manager, for various reasons:
    > *  SecurityException — A runtime exception thrown by Java APIs when a permission check fails. We may deprecate this API for removal at a later date, but for now the impact of doing so would be too high.
  
   -- https://openjdk.org/jeps/411#subject-doas


IMO we'd better keep those during the transition period where we still support JDKs that could possibly be using the security manager.

Reply:
mm, ok. My thoughts here were that since the security exception can only occur in these places when SM is involved (at least according to the javadocs 😃) and we are "removing SM" there's no point in keeping it.